status-im/status-react
# Fixes for the Community Channel avatar component

- Fixes https://github.com/status-im/status-mobile/issues/16332
- Fixes https://github.com/status-im/status-mobile/issues/16327

### Summary

Another step on the journey to fix the Community overview screen. This time around, I'm fixing:

- [x] Community channels may not have an emoji, and in such cases, it should fall back to a single letter. The component was recently updated in [Figma](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=399-20709&mode=design&t=UZYVQAGns9IVRxew-0) after this concern was brought up to designers.
- [x] The default (fallback) color to be used in the list of community channels should be the **community color,** not a default `primary-50` as was before. It's a subtle change because the colors are rendered at 10% transparency.
- [x] Rewrote the `channel-avatar` component to follow guidelines. 

### Examples

<table>
  <thead>
    <tr>
        <th>Before</th>
        <th>After</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="https://github.com/status-im/status-mobile/assets/46027/2926f245-f184-42eb-88dc-9eb3c81aa3ac" width="350" /></td>
      <td><img src="https://github.com/status-im/status-mobile/assets/46027/80893036-99d5-40b9-95b2-5fd6f06fc718" width="350" /></td>
    </tr>
    <tr>
      <td><img src="https://github.com/status-im/status-mobile/assets/46027/47d740df-b299-462c-b305-3808c788b1ff" width="350" /></td>
      <td><img src="https://github.com/status-im/status-mobile/assets/46027/93a8ddd4-aa47-45e1-848d-acab47fb7c6d" width="350" /></td>
    </tr>
  </tbody>
</table>

### Quo component

<img src="https://github.com/status-im/status-mobile/assets/46027/ba90e1fc-abc9-4882-854c-303952ebd646" width="400" />

<img src="https://github.com/status-im/status-mobile/assets/46027/4fe9b912-8ecf-419f-ae7d-dd41c7ceeb61" width="400" />

### How to test

- Open a community with lots of channels, ideally with a mix of custom colors.
- Check the quo2 preview `avatar` > `channel-avatar`.

### Resources

- [Design System > Avatars > Channel Avatar](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=399-20709&mode=design&t=vt6rTZgV5rgSqgfO-0)
- [Communities for Mobile > Channels](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=12697-455084&mode=design&t=urMJx8grNxPPA59i-0)

status: ready
